spark/src/main/scala/geotrellis/spark/pyramid/Pyramid.scala

```diff
@@ -27,9 +31,44 @@ import geotrellis.vector.Extent
 
 import org.apache.spark.Partitioner
 import org.apache.spark.rdd._
+import org.apache.spark.storage.StorageLevel
+import spray.json._
 
 import scala.reflect.ClassTag
 
+case class Pyramid[
+    K: SpatialComponent: ClassTag,
+    V <: CellGrid: ClassTag: ? => TilePrototypeMethods[V]: ? => TileMergeMethods[V],
+    M: Component[?, LayoutDefinition]: Component[?, Bounds[K]]
+](levels: Map[Int, RDD[(K, V)] with Metadata[M]]) {
+  def apply(level: Int): RDD[(K, V)] with Metadata[M] = levels(level)
+
+  def level(l: Int): RDD[(K, V)] with Metadata[M] = levels(l)
+
+  def lookup(zoom: Int, key: K): Seq[V] = levels(zoom).lookup(key)
+
+  def minZoom = levels.keys.min
+  def maxZoom = levels.keys.max
+
+  def persist(storageLevel: StorageLevel) = levels.mapValues{ _.persist(storageLevel) }
+
+  def write(
+    layerName: String,
+    writer: LayerWriter[LayerId],
+    keyIndexMethod: KeyIndexMethod[K]
+  )(
+    implicit arcK: AvroRecordCodec[K],
```

Comment:
Interesting how would it work in the client code.

Reply:
Not sure I understand what is meant by this comment...